src/main/java/de/rwth/idsg/steve/config/SecurityConfiguration.java

```diff
@@ -86,6 +86,27 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                         "/WEB-INF/views/**" // https://github.com/spring-projects/spring-security/issues/13285#issuecomment-1579097065
                     ).permitAll()
                     .requestMatchers(prefix + "/**").hasAuthority("ADMIN")
+                    .requestMatchers(prefix + "/home").hasAnyAuthority("USER", "ADMIN")
+                     // webuser
+                    .requestMatchers(prefix + "/webusers").hasAnyAuthority("USER", "ADMIN")
+                    .requestMatchers(prefix + "/webusers" + "/details/**").hasAnyAuthority("USER", "ADMIN")
+                    // users
+                    .requestMatchers(prefix + "/users").hasAnyAuthority("USER", "ADMIN")
+                    .requestMatchers(prefix + "/users" + "/details/**").hasAnyAuthority("USER", "ADMIN")
+                     //ocppTags
+                    .requestMatchers(prefix + "/ocppTags").hasAnyAuthority("USER", "ADMIN")
+                    .requestMatchers(prefix + "/ocppTags" + "/details/**").hasAnyAuthority("USER", "ADMIN")
+                     // chargepoints
+                    .requestMatchers(prefix + "/chargepoints").hasAnyAuthority("USER", "ADMIN")
```

Comment:
As user can't create, update ou delete charpoints (and some others), the UI should be updated (at least remove delete and update button)

Reply:
> As user can't create, update ou delete charpoints (and some others)

i will play devil's advocate (as setup for my next argument): why not?

your argument has a very specific understanding/model of what a user and admin should do or do not. i find it really hard to make decisions in steve about permissions and roles matrix because each customer/CPO/company would have different requirements depending on their organisation structure.

i have no solution for this. i dont think we can make some decisions (in this generic version of steve) that will make people happy.

Reply:
the perfectionist in me says.. we should make the permissions and roles configurable in web UI or something, but this would be a big undertaking.

Reply:
Sure, why not?

Let me share my own experience where people (administrative persons but technical operators too) unintentionally break things because they can. 

But what if we could give them a way to only "read" data and help drivers?

At the same time, "users" could be drivers in different contexts. 

In this vision, users become helpful companions for drivers/them, providing information and simple commands that won't cause any damage. Meanwhile, admins will be in charge of managing the network of stations.

Now, the question arises: how would we differentiate between admins and users? Can you imagine any noticeable differences in the way they are currently implemented?

To the perfectionist: I think it is recommended that implementation of peripheral functionalities, such as the one in question or even IAM, be delegated to specialized software stacks. This ensures that the primary objective of the software solution remains the central focus, while benefiting from the seamless integration of a well-established stack that effectively addresses the requirement.
However, it is important to note that implementing this way will require additional effort and may introduce complexity to the existing system. This could potentially impact the ops part.

Reply:
Actually the button are still there, but USER landing on the NoAccess site. If someone with more *jsp experience can hide them, I would welcome this!

Reply:
I can do some ui improvments but @geokay must define first what is expected for USER and ADMIN.

Reply:
Another option can be ADMIN, SUPERVISOR, USER.
With 
USER = readonly, except users,
SUPERVISOR = read+write, except users,
ADMIN = read+write all

Wdyt @goekay ? Please advice.

Reply:
i wrote this in an [earlier comment](https://github.com/steve-community/steve/pull/1544#discussion_r1720797188), and i still think that way: probably, we should not make any decisions about this. 

we can create the basic framework (i.e. the preparation for decisions) and ability to create roles. what these roles can do and cannot do should be left to each user of steve.

there are so many aspects and functionalities, i.e. master data management (read and write) and various ocpp operations. i find it hard to come up with one matrix that will be _the_ solution.

otherwise, we will get questions/issues like: why cant my USER role do X?

Reply:
In the current state of the code creating roles is possible by adapting the enum. 
Changing the privileges can be done by adapting the SecurityFilterChain, which of course could be tricky (https://github.com/steve-community/steve/pull/1544#issuecomment-2314552615). 
Are comments in the source code sufficient? 

So for the basic framework solution configuration I suggest:
User --> allowed to see all website/data
Admin --> allowed to see all websites and is allowed to change the data
[User, Admin] is commented out in the enum. I would like to keep it as a hint that multiple roles are technically possible